cosmos/x/evm/plugins/state/genesis.go

```diff
@@ -85,6 +89,23 @@ func (p *plugin) ExportGenesis(ctx sdk.Context, ethGen *core.Genesis) {
 
 		account.Code = p.GetCode(address)
 		account.Balance = p.GetBalance(address)
+		account.Nonce = p.GetNonce(address)
+		ethGen.Alloc[address] = account
+
+		return false
+	})
+
+	// Iterate Code and set the genesis accounts.
+	p.IterateCode(func(address common.Address, codeHash common.Hash) bool {
+		account, ok := ethGen.Alloc[address]
+		if !ok {
+			account = core.GenesisAccount{}
+		}
+		account.Code = p.GetCode(address)
+		account.Nonce = p.GetNonce(address)
+		if account.Balance == nil {
+			account.Balance = big.NewInt(0)
+		}
```

Comment:
just ask; nil check here is mandatory? any issue without it?

Reply:
If balance doesn't exist, `InitGenesis` will fail at `p.SetBalance(address, account.Balance)`. Because GenesisAccount must have balance.
So i considered about checking balance when InitGenesis, but I thought handling balance in ExportGenesis would be more reasonable to handle it in InitGenesis.

Reply:
hmm I thought that it'd be better to handle it in `InitGenesis` for consistency because there are already many nil checks in `InitGenesis`.

Reply:
okay thx :)